### DISCUSSION TITLE
CheckMK Agent on Flatcar

### DISCUSSION THREAD

**Discussion by MarcoGriep88** [flatcar/Flatcar#1494] [Q&A]
Hi there,

got my terraform and flatcar running with docker containers. I want to monitor my system and also provide the installation of an checkmk agent on the server (or node_exporter for prometheus, i dont care). Is it somehow possible to install the rpm/deb package? I mean... There is no package manager right?

i already provided the deb file in the files section and download it from an S3 Storage.

    - path: /home/core/checkmk.deb
      contents:
        source: https://flatcar-######.s3-eu-#####-1.ionoscloud.com/#######/check-mk-agent_2.3.0p8-1_all.rpm
        verification:
          hash: sha512-#####

But i dont know how to run it so check_mk agent is running.

Best Regards
Marco

**Comment by tormath1** [comment]
Hi,

As you mentioned, there is no package manager on Flatcar and `.deb` can only be installed using `dpkg` or `apt` tools, which are not available on Flatcar by design.

I would recommend to run the agent as a Docker container by following this documentation: https://docs.checkmk.com/latest/en/introduction_docker.html - with Butane, you can define a systemd unit that setup the container.

Similar to this nginx configuration: https://www.flatcar.org/docs/latest/installing/#writing-your-first-config-and-testing-it-locally-in-a-qemu-vm

Let us know how it goes!

**Comment by MarcoGriep88** [comment]
Hi @tormath1 , thanks for your fast reply.

Unfortunatly there is no official docker Image for the Agent, and the docs are only referring to the Server install. There are old inofficial agent Images but they Look unmaintained (Last Update 4 years ago). I will try My luck with node Exporter for Prometheus because checkmk Server can also query Prometheus Sources, there is an single binary File for Linux i can Register as Service and also an docker Image. I will try It this way.

Thanks alot. Monitoring My system ist quite Important for me :)

I will share My Butan File as soon as i got It Up and running.

Best regards

**Reply by ader1990** [reply]
You could build a Docker image from the latest Ubuntu image plus the .deb installation, and then use it with Docker on Flatcar, as @tormath1 suggested.
